cpp/src/arrow/record_batch.h

```diff
@@ -310,7 +310,20 @@ class ARROW_EXPORT RecordBatchReader {
   /// \brief Read the next record batch in the stream. Return null for batch
   /// when reaching end of stream
   ///
-  /// \param[out] batch the next loaded batch, null at end of stream
+  /// example:
+  /// ```
+  /// std::shared_ptr<RecordBatch> batch;
+  /// while (true) {
+  ///   ARROW_RETURN_NOT_OK(reader->ReadNext(&batch));
+  ///   if (batch == nullptr) {
```

Comment:
FYI: I don't object `== nullptr` but I like `!batch` style.

```suggestion
  ///   if (!batch) {
```